src/main/java/com/spotify/reaper/service/SegmentRunner.java

```diff
@@ -298,6 +307,35 @@ public void handle(int repairNumber, ActiveRepairService.Status status, String m
   }
 
   /**
+   * Attempts to clear snapshots that are possibly left behind after failed repair sessions.
+   */
+  @VisibleForTesting
+  protected void maybeClearSnapshots(String message) {
+    String keyspace = repairUnit.getKeyspaceName();
+    String repairId = parseRepairId(message);
+    if (repairId != null) {
+      for (String involvedNode : potentialCoordinators) {
+        try {
+          JmxProxy jmx = new JmxConnectionFactory().connect(involvedNode);
+          jmx.clearSnapshot(repairId, keyspace);
```

Comment:
We should log successful snapshot removal if we can. Will it throw when the directory is not there? In that case, not throwing would mean that doing this has accomplished something, which is worth logging.